python/cuml/cuml/ensemble/randomforest_common.pyx

```diff
@@ -72,6 +72,31 @@ class BaseRandomForestModel(UniversalBase):
 
     classes_ = CumlArrayDescriptor()
 
+    @classmethod
+    def _criterion_to_split_criterion(cls, criterion):
```

Comment:
It looks like this is manually called in the `_hyperparam_translator` by all subclasses of `BaseRandomForestModel`. Wouldn't it be cleaner to define `_hyperparam_translator` on this class with this logic inlined, then subclasses will get this for free with the `super()...` call even if they redefine `_hyperparam_translator` (which they all do).

Also, slightest of preferences to move this down a bit - methods defined before `__init__`/`__new__` weird me out :).

Reply:
I thought about this as well, and I have a slight preference for keeping the actual translation more local to the place its actually used, leaving this as just a utility function for string remapping. In general, we've got too much hard-to-follow inheritance going on in this class structure, so I shied away from adding anything more. This would be a good one to restructure fully in a follow-up PR.

As for the order of methods, I generally put static methods first, then class methods, then `__new__`, then `__init__`. Not religious about it, but let's leave it as is until/unless we decide to adopt a standard convention.